File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums, target):
        # Dictionary to map number -> index
        seen = {}

        # Loop through the list
        for i, num in enumerate(nums):
            # Calculate the number we need
            needed = target - num
            
            # If the needed number was already seen, return result
            if needed in seen:
                return [seen[needed], i]
            
            # Otherwise, store current number with its index
            seen[num] = i

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> seen;
        for (int i=0;i<nums.size();i++){
            int x=target-nums[i];

            if (seen.find(x)!=seen.end()){
                return {seen[x],i};
            }else{
                seen[nums[i]]=i;
            }
        }
        return{};
    }
};